app/src/main/java/com/github/sdp/tarjetakuna/database/DBMagicCard.kt

```diff
@@ -15,26 +15,36 @@ data class DBMagicCard(
     val code: String,
     val number: Int,
 ) {
-    constructor(card: MagicCard, possession: CardPossession) : this(
+    constructor(card: MagicCard) : this(
         card = Gson().toJson(card),
-        possession = possession,
+        possession = CardPossession.NONE,
         lastUpdate = System.currentTimeMillis(),
         code = card.set.code,
         number = card.number,
     )
 
-    constructor(card: MagicCard) : this(
+    constructor(card: MagicCard, possession: CardPossession) : this(
         card = Gson().toJson(card),
-        possession = CardPossession.NONE,
+        possession = possession,
         lastUpdate = System.currentTimeMillis(),
         code = card.set.code,
         number = card.number,
     )
 
+    /**
+     * Converts a DBMagicCard to a MagicCard
+     */
     fun toMagicCard(): MagicCard {
         return Gson().fromJson(card, MagicCard::class.java)
     }
 
+    /**
+     * Creates a DBMagicCard with no possession
+     */ //todo: test this
+    fun clearPossession(): DBMagicCard {

```

Comment:
easier to put possession as var no ?

Reply:
What do you mean?

Reply:
well, here to change the possesion, you need to recreate another card. might as well change possession from a val to a var type, so you can simply change it, without recreating a DBMagicCard

Reply:
true, but I'm not sure if its better to keep the immutability of the class.